pkg/apis/cr/v1alpha1/types.go

```diff
@@ -131,6 +132,16 @@ type ActionStatus struct {
 	DeferPhase Phase `json:"deferPhase,omitempty"`
 }
 
+// ActionProgress provides information on the progress of an action.
+type ActionProgress struct {
+	// PercentCompleted is computed by assessing the number of completed phases
+	// against the the total number of phases.
+	PercentCompleted string `json:"percentCompleted"`
+	// LastTransitionTime represents the last date time when the progress status
+	// was received.
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
```

Comment:
As discussed offline, looks like the only difference between my attempts to fix and your PR is the `omitempty` 💡 🤔 
Should we add it to `PercentCompleted` also?